lib/src/legacy/index.ts

```diff
@@ -172,7 +172,9 @@ function convertStringOptions<sync extends 'sync' | 'async'>(
 
   return {
     ...modernOptions,
-    url: options.file ? pathToFileURL(options.file) : undefined,
+    url: options.file
+      ? pathToFileURL(options.file)
+      : new URL(legacyImporterProtocol),
```

Comment:
Legacy API needs to have a `FilesystemImporter('.')` as if it has an `url` even if it does not. Set a special value here to let the new API know that.